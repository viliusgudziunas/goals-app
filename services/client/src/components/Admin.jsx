import React, { useState, useEffect } from 'react';
import axios from 'axios';

import UsersList from './UsersList';
import AddUser from './AddUser';

const Admin = () => {
  const [users, setUsers] = useState([]);
  const [pingFetchUsers, setPingFetchUsers] = useState(true);

  useEffect(() => {
    axios
      .get(`${process.env.REACT_APP_USERS_SERVICE_URL}/users`)
      .then(res => {
        setUsers(res.data.data);
      })
      .catch(err => {
        console.log(err);
      });
  }, [pingFetchUsers]);

  return (
    <div className='container'>
      <br />
      <h1>All Users</h1>
      <hr />
      <AddUser
        pingFetchUsers={pingFetchUsers}
        setPingFetchUsers={setPingFetchUsers}
      />
      <hr />
      <UsersList users={users} />
    </div>
  );
};

export default Admin;
