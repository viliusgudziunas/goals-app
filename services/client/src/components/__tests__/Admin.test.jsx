import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Admin from '../Admin';

describe('Admin', () => {
  it('renders a snapshot properly', () => {
    const tree = renderer.create(<Admin />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const wrapper = shallow(<Admin />);
  it('renders properly', () => {
    const header = wrapper.find('h1');
    expect(header.text()).toBe('All Users');
  });
});
