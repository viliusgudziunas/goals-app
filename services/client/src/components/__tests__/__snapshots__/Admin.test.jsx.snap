// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> should match the snapshot 1`] = `"<div class=\\"container\\"><br/><h1>All Users</h1><hr/><div class=\\"row\\"><form class=\\"col-md-4\\"><div class=\\"form-group\\"><input type=\\"email\\" class=\\"form-control\\" placeholder=\\"Enter an email address\\" id=\\"formEmail\\" value=\\"\\" required=\\"\\"/></div><button type=\\"submit\\" class=\\"btn btn-success btn-block\\">Submit</button></form></div><hr/><div class=\\"row\\"><div class=\\"col\\"><div class=\\"row list-group list-group-flush\\"></div></div></div></div>"`;
